agent/src/main/java/io/pyroscope/javaagent/CurrentPidProvider.java

```diff
@@ -0,0 +1,28 @@
+package io.pyroscope.javaagent;
+
+import sun.management.VMManagement;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class CurrentPidProvider {
+    public static int getCurrentProcessId() {
```

Comment:
Is it possible to use `long pid = ProcessHandle.current().pid();` on Java>=9 ?

Reply:
I got this on correto 17.0.9 win x64
```
java.lang.reflect.InaccessibleObjectException: Unable to make field private final sun.management.VMManagement sun.management.RuntimeImpl.jvm accessible: module java.management does not "opens sun.management" to unnamed module @4bb4de6a
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at io.pyroscope.javaagent.CurrentPidProvider.getCurrentProcessId(CurrentPidProvider.java:17)
	at io.pyroscope.javaagent.JFRProfilerDelegate.start(JFRProfilerDelegate.java:45)
	at io.pyroscope.javaagent.impl.ContinuousProfilingScheduler.startFirst(ContinuousProfilingScheduler.java:96)
	at io.pyroscope.javaagent.impl.ContinuousProfilingScheduler.start(ContinuousProfilingScheduler.java:44)
	at io.pyroscope.javaagent.PyroscopeAgent.start(PyroscopeAgent.java:51)
	at App.main(App.java:21)
	```

Reply:
lets add a unit test for this function and run it in gh actions

Reply:
can't we do it at runtime? or maybe also with reflection? 
I dont like the idea to have multiple jars.
I dont realy mind how it works, it should just work and currently it throws exception.

Reply:
multirelease is a single jar, which holds different versions of classes for different JDK versions, https://openjdk.org/jeps/238, so it is single jar :)

Reply:
> multirelease is a single jar, which holds different versions of classes for different JDK versions, https://openjdk.org/jeps/238, so it is single jar :)

Nice. Did not know about it. This should do the trick.